DiamondLightSource/python-artemis
# 973 speed up test pin centring

Fixes #973 

- Identified test functions `test_pin_tip_starting_near_negative_edge_doesnt_exceed_limit` and `test_pin_tip_starting_near_positive_edge_doesnt_exceed_limit` that took the most time to run
- Created mock object and mock generator to replace call to `yield from trigger_and_return_pin_tip(oav.mxsc.pin_tip)` in two test functions that took the most time
- Added `.side_effect` to mock generators

### To test:
1. Run tests in _test_pin_tip_centring.py_ in **main branch** and compare how long they take with **973_speed_up_test_pin_centring branch**
2. Confirm branch is faster using the Testing feature in VSCode
